src/Compilers/CSharp/Portable/Binder/RefSafetyAnalysis.cs

```diff
@@ -82,6 +84,11 @@ private static bool InUnsafeMethod(Symbol symbol)
         private Dictionary<LocalSymbol, (uint RefEscapeScope, uint ValEscapeScope)>? _localEscapeScopes;
         private Dictionary<BoundValuePlaceholderBase, uint>? _placeholderScopes;
         private uint _patternInputValEscape;
+#if DEBUG
+        private const int MaxTrackVisited = 100; // Avoid tracking if too many expressions.
+        private readonly HashSet<BoundExpression> _visited = new HashSet<BoundExpression>();
+        private bool _trackVisited = true;
```

Comment:
nit: we could do without this bool. Just set `_visited` to `null` when we want to disable tracking